From 6bbb2f68cda9854995fc8d726f93afc7350519cd Mon Sep 17 00:00:00 2001
From: Willy Tarreau <w@1wt.eu>
Date: Mon, 9 Dec 2013 17:14:23 +0100
Subject: [PATCH 273/277] MINOR: session: report lack of resources using the
 new stream-interface's error code

Let's now use SI_ET_CONN_RES to report lack of resources instead of
SO_ET_CONN_OTHER with a handcrafted code.
---
 src/session.c | 5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git a/src/session.c b/src/session.c
index 75ac04c..86dcb84 100644
--- a/src/session.c
+++ b/src/session.c
@@ -1172,13 +1172,12 @@ static void sess_prepare_conn_req(struct session *s, struct stream_interface *si
 			 * error code to ignore the ERR_LOCAL which is not a
 			 * real error.
 			 */
-			s->flags = (s->flags & ~SN_ERR_MASK) | SN_ERR_RESOURCE;
-			s->flags = (s->flags & ~SN_FINST_MASK) | SN_FINST_C;
+			s->flags &= ~(SN_ERR_MASK | SN_FINST_MASK);
 
 			si_shutr(si);
 			si_shutw(si);
 			si->ob->flags |= CF_WRITE_ERROR;
-			si->err_type = SI_ET_CONN_OTHER;
+			si->err_type = SI_ET_CONN_RES;
 			si->state = SI_ST_CLO;
 			if (s->srv_error)
 				s->srv_error(s, si);
-- 
1.8.1.5

